apps/ringtones/js/tone_player.js

```diff
@@ -93,11 +92,18 @@ TonePlayer.prototype = {
   /**
    * Stop playing the current tone (if any). If we start playing it again,
    * setTone will reset the current time to 0. Note: even though we're stopping,
-   * we don't want to clear the player's src URL, since we might need that to
-   * restart playback of the same song.
+   * we don't want to clear the player's src URL, since we'd end up forgetting
+   * if we had validated the tone (we need that information when handling a pick
+   * activity).
+   *
+   * @param {Boolean} close true if we should also deactivate "exclusive mode"
+   *   (read: close the audio channel associated with the tones).
    */
-  stop: function() {
+  stop: function(close) {
     this._player.pause();
+    if (close) {
```

Comment:
If I'm understanding your code correctly, I think you already call setExclusiveMode(true) if this tone is played again. So IIUC, you could just unconditionally call setExclusiveMode(false) here, and then you wouldn't need to add the argument.  And that would simplify your patch a lot.